Loop/Managers/DeviceDataManager.swift

```diff
@@ -307,35 +285,39 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
         }
     }
 
-    private func readAndUpdatePumpData(completion: (error: ErrorType?) -> Void) {
+    /**
+     Read the pump's current state, including reservoir and clock
 
+     - parameter completion: A closure called after the command is complete. This closure takes a single Result argument:
+        - Success(status, date): The pump status, and the resolved date according to the pump's clock
+        - Failure(error): An error describing why the command failed
+     */
+    private func readPumpData(completion: (Either<(status: RileyLinkKit.PumpStatus, date: NSDate), ErrorType>) -> Void) {
         guard let device = rileyLinkManager.firstConnectedDevice, let ops = device.ops else {
-            completion(error: LoopError.ConfigurationError)
+            completion(.Failure(LoopError.ConfigurationError))
             return
         }
 
-        ops.readPumpStatus({ (result) in
+        ops.readPumpStatus { (result) in
             switch result {
             case .Success(let status):
-                self.latestPolledPumpStatus = status
-
                 status.clock.timeZone = ops.pumpState.timeZone
                 guard let date = status.clock.date else {
                     self.logger.addError("Could not interpret pump clock: \(status.clock)", fromSource: "RileyLink")
-                    completion(error: LoopError.ConfigurationError)
+                    completion(.Failure(LoopError.ConfigurationError))
                     return
                 }
 
-                self.updateReservoirVolume(status.reservoir, atDate: date, withTimeLeft: nil)
```

Comment:
The read + update cases needed to be decoupled originally for the bolus case, since `updateReservoirVolume` triggers history polling and looping.

short-term, it seemed best to revert the behavior until history-based insulin dosage is ready, for which the device-trigger web will need to be pliable yet again.